packages/tizen_app_manager/tizen/src/application_utils.cc

```diff
@@ -4,145 +4,194 @@
+    LOG_ERROR("get icon path error! : %s", get_error_message(ret));
     // because some system app and service app don't have icon,
     // just print error log, and pass it
-    LOG_ERROR("get icon path error! : %s", get_error_message(ret));
+  } else {
+    value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
+        "iconPath", std::string(icon_path)));
   }
 
-  ret = app_info_get_exec(app_info, &exec_path);
+  if (icon_path) {
+    free(icon_path);
+  }
+}
+
+void GetExecutablePath(app_info_h app_info, flutter::EncodableMap &value) {
+  char *exec_path = nullptr;
+
+  int ret = app_info_get_exec(app_info, &exec_path);
   if (ret != APP_MANAGER_ERROR_NONE || exec_path == nullptr) {
     LOG_ERROR("get exec path error! : %s", get_error_message(ret));
-    goto cleanup;
+    if (exec_path) {
+      free(exec_path);
+    }
+    throw ret;
   }
+  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
+      "executablePath", std::string(exec_path)));
+  free(exec_path);
+}
+
+void GetSharedResourcePath(char *app_id, flutter::EncodableMap &value) {
+  char *shared_res_path = nullptr;
 
-  ret = app_manager_get_shared_resource_path(app_id, &shared_res_path);
+  int ret = app_manager_get_shared_resource_path(app_id, &shared_res_path);
   if (ret != APP_MANAGER_ERROR_NONE || shared_res_path == nullptr) {
     LOG_ERROR("get shared resource path error! : %s", get_error_message(ret));
-    goto cleanup;
+    if (shared_res_path) {
+      free(shared_res_path);
+    }
+    throw ret;
   }
+  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
+      "sharedResourcePath", std::string(shared_res_path)));
+  free(shared_res_path);
+}
+
+void GetIsNoDisplay(app_info_h app_info, flutter::EncodableMap &value) {
+  bool no_display = false;
 
-  ret = app_info_is_nodisplay(app_info, &no_display);
+  int ret = app_info_is_nodisplay(app_info, &no_display);
   if (ret != APP_MANAGER_ERROR_NONE) {
     LOG_ERROR("app_info_is_nodisplay error! : %s", get_error_message(ret));
-    goto cleanup;
+    throw ret;
   }
+  value.insert(std::pair<flutter::EncodableValue, flutter::EncodableValue>(
+      "isNoDisplay", no_display));
+}
+
+void GetForEachMetadata(app_info_h app_info, flutter::EncodableMap &value) {
+  flutter::EncodableMap metadata;
 
-  ret = app_info_foreach_metadata(app_info, AppMetaDataCB, (void *)&metadata);
+  int ret = app_info_foreach_metadata(
+      app_info,
+      [](const char *key, const char *value, void *user_data) {
+        if (key != nullptr) {
+          flutter::EncodableMap *metadata =
+              static_cast<flutter::EncodableMap *>(user_data);
+          metadata->insert(
+              std::pair<flutter::EncodableValue, flutter::EncodableValue>(
+                  std::string(key), std::string(value)));
+          LOG_INFO("AppMetaDataCB key: %s, value: %s", key, value);
+        }
+        return true;
+      },
+      (void *)&metadata);
```

Comment:
Please don't use C-style cast.

Reply:
It has been modified to use static_cast.
`(void *)&metadata` -> `static_cast<void *>(&metadata)`